src/__init__.py

```diff
@@ -1,242 +1,76 @@
-        white=(255,255,255)
-        yellow=(255,255,0)
-        black=(10,10,10)
-        # Draw Austrian flag
-        AT=[]
-        AT=[white]*40
-        # TOP
-        for i in range(0,9):
-            AT[i]=red
-        for i in range(32,40):
-            AT[i]=red
-        # MID
-        for i in range(9,17):
-            AT[i]=white
-        for i in range(24,32):
-            AT[i]=white
-        # BOT
-        for i in range(17,25):
-            AT[i]=red
-        DE=[]
-        DE=[white]*40
-        # TOP
-        for i in range(0,9):
-            DE[i]=black
-        for i in range(32,40):
-            DE[i]=black
-        # MID
-        for i in range(9,17):
-            DE[i]=red
-        for i in range(24,32):
-            DE[i]=red
-        # BOT
-        for i in range(17,25):
-            DE[i]=yellow
-        # Draw AustrianHungarian flag
-        ATHU=[]
-        ATHU=[white]*40
-        # TOP
-        for i in range(0,9):
-            ATHU[i]=yellow
-        for i in range(32,40):
-            ATHU[i]=yellow
-        # MID
-        for i in range(9,17):
-            ATHU[i]=black
-        for i in range(24,32):
-            ATHU[i]=black
-        for i in range(17,25):
-            ATHU[i]=black
-        return [ AT, ATHU, DE ]
-
-    def pulse(self, brightness):
-        if self._pulse_inc:
-            brightness = brightness + self._pulse_by
-        else:
-            brightness = brightness - self._pulse_by
-        if brightness <= 0:
-            self._pulse_inc = not self._pulse_inc
-            return 0
-        if brightness >= 255:
-            self._pulse_inc = not self._pulse_inc
-            return 255
-        return brightness
-
-    def toggle_pulse(self):
-        self._pulse=not self._pulse
-        if self._pulse:
-            self._pulse_inc=True
-            self._pulse_by=5
-            leds.set_brightness(0)
-
-    def set_ironman(self):
-        self._flag = None
-        self._ironman = True
-        self._cthulhu = False
-
-    def set_cthulhu(self):
-        self._flag = None
-        self._ironman = False
-        self._cthulhu = True
-
-    def set_flag(self):
-        self._selected_flag = (self._selected_flag + 1) % len(self._flags)
-        self._flag = self._flags[self._selected_flag]
-        self._ironman = False
-        self._cthulhu = False
+        self.input = InputController()
+        self.rot, self.flag_id, self.picture_id, self.brightness = 0, 0, 0, 100
+        self.petal_last_frame = [False] * 10
+        files = os.listdir(PATH)
+        self.img_files = [PATH + p for p in files if p.endswith(".png") or p.endswith(".jpg")]
+        self.audio_files = [self.blm.new(bl00mbox.patches.sampler, PATH + p) for p in files if p.endswith(".wav")]
+
+    def update_leds(self):
+        colors = COLORS[self.flag_id]
+        for i in range(NR_LEDS):  # set flag
+            i_led = int(i + self.rot * NR_LEDS) % NR_LEDS
+            i_color = int(i / NR_LEDS * len(colors))
```

Comment:
Wenns funktioniert, fein - ich bin wohl zuwenig Programmierer um das aufs Hinschaun zu verstehen.

Reply:
du gehst über alle 40 LEDS, und berechnest pro led auf welchem led und in welcher farbe du bist. da der offset (based on rot) const ist gehst du trotzdem über alle 40 leds.